mapreduce/master.js

```diff
@@ -0,0 +1,7 @@
+"use strict";
+
+module.exports.master = (event, context, callback) => {
```

Comment:
So how does this work exactly? I'm not familiar with this type of event. Any change to dynamoDb will trigger master? Or just the creation of a new job? And it looks like it could batch jobs (`event.Records`)?

Reply:
And what are you imagining this handler will do?

Reply:
It gets triggered anytime a row is changed in dynamo db. It uses dynamo db streams https://aws.amazon.com/blogs/database/dynamodb-streams-use-cases-and-design-patterns/. 

It can batch it is triggered based on batch size. I set it to 1 right now.

I envision this being the "master" for the map reduce job. It will fetch the data set ( the path will be loaded from the job table in dynamo db). Split up the dataset into chunks.

It will trigger the mappers for each chunk of the dataset. It will take the output of the map, then trigger the shuffle phase. After shuffle is done it coordinate the reducers and then write output.

Reply:
That level of granularity isn't supported we will have to filter that out in code.

Reply:
It looks like it should be possible according to AWS, but I'm having a hard time finding out what exactly needs to change. Here's a tutorial: https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Streams.Lambda.Tutorial.html#Streams.Lambda.Tutorial.CreateTrigger

Reply:
https://stackoverflow.com/questions/50550849/configure-dynamodb-stream-to-invoke-lambda-function-only-on-delete

https://stackoverflow.com/questions/41112362/configure-dynamodb-stream-trigger-with-insert-only

I can't seem to find out how without using it in code

Reply:
never mind, it looks like they just check `record.eventName == 'INSERT'`